torch/lib/c10d/ProcessGroupGloo.cpp

```diff
@@ -363,6 +364,12 @@ void ProcessGroupGloo::SendWork::wait() {
   if (exception_) {
     std::rethrow_exception(exception_);
   }
+  return sendCompleted;
+}
+
+void ProcessGroupGloo::SendWork::abort() {
+  std::unique_lock<std::mutex> lock(mutex_);
+  buffer_->abortWaitSend();
```

Comment:
should we also set `aborted_ = true` here for completeness?

Reply:
I don't feel it's the right thing to do. 
I think If a sub class doesn't use a field in the base class, just don't bother to use and modify it.